awx/api/fields.py

```diff
@@ -101,8 +101,15 @@ def __init__(self, **kwargs):
         super(DeprecatedCredentialField, self).__init__(**kwargs)
 
     def to_internal_value(self, pk):
+        if isinstance(pk, str):
+            self.fail('invalid')
         try:
+            pk = int(pk)
+        except (ValueError):
+            self.fail('invalid')
+        try:
+            pk = int(pk)
```

Comment:
this additional try-except block is all I was looking for.

The `pk = int(pk)` line here is a duplicate, so you'll want to remove that, as well as the () around the object does not exist exception below.